# Question 2 Analysis 1
- with NYT archive API download all news in year 1993
- find name keywords of every article in this year
- try to find the most important person in this year
- generate a CSV file showing these important people

In [26]:
import glob
import os
import json
from pathlib import Path
from collections import Counter

In [27]:
# prepare the path to news in year 1993
p = Path(os.getcwd())
archive_path = str(p.parent) + '/data/archive/1993/*'

In [28]:
# get the persons in keyword of those news and add them to person_list
files = glob.glob(archive_path)
person_list = []
for file in files:
    with open(file, 'r', encoding='utf-8', errors='ignore') as f:
        python_data = json.load(f)
        docs = python_data['response']['docs']
        for doc in docs:
            if 'keywords' in doc:
                keywords = doc['keywords']
                for key in keywords:
                    if key['name'] == 'persons':
                        person_list.append(key['value'])
print(len(person_list))

59468


In [29]:
# count the frequency of those names and order them
name_counter = Counter()
for name in person_list:
    name_counter.update([name])
for name, time in name_counter.most_common(10):
    print(name+': '+ str(time))

CLINTON, BILL: 3863
DINKINS, DAVID N: 787
GIULIANI, RUDOLPH W: 473
YELTSIN, BORIS N: 354
BUSH, GEORGE: 342
CUOMO, MARIO M: 277
CLINTON, HILLARY RODHAM: 232
FLORIO, JAMES J: 219
WHITMAN, CHRISTINE TODD: 206
CHRISTOPHER, WARREN M: 157


In [30]:
# write those persons into CSV file in popularity-descending order
import csv
output_path = 'ana_1/important_persons.csv'
with open(output_path, 'w') as outcsv:
    writer = csv.DictWriter(outcsv, fieldnames = ["PERSON","TIMES_OF_OCCURANCE"])
    writer.writeheader()
    for name, time in name_counter.most_common(50):
        writer.writerow({'PERSON': name,'TIMES_OF_OCCURANCE': str(time)})


# Conclusion
- As Bill Clinton showed up 3863 times in keywords in year 1993, we have plenty of evidence to say he is the most important (or controversial) person in that year
- Just behind Bill Clinton, David Dinkins (106th Mayor of New York City, from 1990 to 1993) and Rudolph Giuliani (107th Mayor of New York City, from 1994 to 2001) are also reasonable to be popular on NYT news